app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_list = []
+    for person in people:
+        new_list.append(Person(name=person["name"], age=person["age"]))
+    for index, person in enumerate(people):
+        if "wife" in person and person["wife"] is not None:
```

Comment:
You can use .get method in dict and simplify the condition
The same with husband